algorithmic_efficiency/workloads/mnist/mnist_pytorch/workload.py

```diff
@@ -101,18 +115,31 @@ def build_input_queue(self,
                         data_rng,
                         split: str,
                         data_dir: str,
-                        global_batch_size: int):
-    return self._build_dataset(data_rng, split, data_dir, global_batch_size)
+                        global_batch_size: int) -> Dict[str, Any]:
+    it = self._build_dataset(data_rng, split, data_dir, global_batch_size)
+    for batch in it:
+      if isinstance(batch, dict):
+        inputs = batch['inputs']
+        targets = batch['targets']
+      else:
+        inputs, targets = batch
```

Comment:
The custom [`cycle`](https://github.com/mlcommons/algorithmic-efficiency/blob/main/algorithmic_efficiency/data_utils.py#L27-L39) function returns a dict, but is only used for training here, because for testing `itertools.cycle` is used, which catches the whole dataset in memory, which shouldn't happen during training even if the size of the dataset is no issue because it wil not re-shuffle. But I don't think it is less efficient to just use my custom cycle function also for the test set, so we can get rid of the if-statement (and it is consistent with e.g. the ImageNet workload).